transceivable.go

```diff
@@ -127,6 +129,8 @@ func (t *transceivable) Close() (err error) {
 		if t.settings.OnClosed != nil {
 			t.settings.OnClosed(ExplicitClosing)
 		}
+
+		t.wg.Wait()
 	}
 	return
 }
```

Comment:
Please update this function, reuse your nearly created `closing`
```go
func (t *transceivable) Close() error {
    t.closing(ExplicitClosing)
    return  nil
}
```

Reply:
Some tests were failing after recommended change. So I added a returned error to the new closing () func.
```golang
func (t *transceivable) Close() (err error) {
	return t.closing(ExplicitClosing)
}

func (t *transceivable) closing(state State) (err error) {
	if atomic.CompareAndSwapInt32(&t.aliveState, Alive, Closed) {
		...
		// close underlying conn
		err = t.conn.Close()
		...
	}
	return
}
```